packages/components/input-number/src/input-number.vue

```diff
@@ -161,54 +165,50 @@ export default defineComponent({
       }
       return precision
     }
-    const _increase = (val: number) => {
-      if (!isNumber(val)) return data.currentValue
-      const precisionFactor = 10 ** numPrecision.value
-      // Solve the accuracy problem of JS decimal calculation by converting the value to integer.
-      val = isNumber(val) ? val : Number.NaN
-      return toPrecision(
-        (precisionFactor * val + precisionFactor * props.step) / precisionFactor
-      )
-    }
-    const _decrease = (val: number) => {
+    const ensurePrecision = (val: number, coefficient: 0 | 1 | -1 = 1) => {
       if (!isNumber(val)) return data.currentValue
-      const precisionFactor = 10 ** numPrecision.value
       // Solve the accuracy problem of JS decimal calculation by converting the value to integer.
       val = isNumber(val) ? val : Number.NaN
-      return toPrecision(
-        (precisionFactor * val - precisionFactor * props.step) / precisionFactor
-      )
+      return toPrecision(val + props.step * coefficient)
     }
     const increase = () => {
       if (inputNumberDisabled.value || maxDisabled.value) return
       const value = props.modelValue || 0
-      const newVal = _increase(value)
+      const newVal = ensurePrecision(value)
       setCurrentValue(newVal)
     }
     const decrease = () => {
       if (inputNumberDisabled.value || minDisabled.value) return
       const value = props.modelValue || 0
-      const newVal = _decrease(value)
+      const newVal = ensurePrecision(value, -1)
       setCurrentValue(newVal)
     }
-    const setCurrentValue = (newVal: number | string) => {
-      const oldVal = data.currentValue
-      if (props.stepStrictly) {
-        const stepPrecision = getPrecision(props.step)
-        const precisionFactor = 10 ** stepPrecision
-        newVal =
-          (Math.round(newVal / props.step) * precisionFactor * props.step) /
-          precisionFactor
+    const verifyValue = (
+      value: number | string | undefined,
+      update?: boolean
+    ) => {
+      let newVal = Number(value)
```

Comment:
**Suggestion:**

```ts
const { max, min, step, precision } = props // move then at the top of this method to avoid duplication 
```

Reply:
If we do what I suggested, we wouldn't need to use `newVal = newVal > props.max ? props.max : props.min` it would be much shorter.